script.rb

```diff
@@ -0,0 +1,172 @@
+# rubocop:disable all
+
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    var = to_a
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < var.length
+      yield(var[index])
+      index += 1
+    end
+    var
+  end
+
+  def my_each_with_index
+    var = to_a
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < var.length
+      yield(var[index], index)
+      index += 1
+    end
+    var
+  end
```

Comment:
Here, the test shows:

![image](https://user-images.githubusercontent.com/2739245/94213810-75544900-fe9d-11ea-9b03-921e0b3fa2b3.png)

Again, the return value for a range should be a range, not an array.